animal.go

```diff
@@ -3,13 +3,19 @@ package animal
 import (
 	"bufio"
 	"fmt"
-	"os"
+	"io"
 )
 
 type game struct {
 	Running bool
 }
 
+// Data struct for the game
+type Data struct {
```

Comment:
I see your reasoning here, but it probably isn't really that useful to define a struct like this. The question string and the reader we read the response from are essentially unrelated things, so we would probably pass them as separate parameters to the function.

Indeed, we might even make the reader a field on the `game` struct, so that you don't need to supply `os.Stdin` for the real game; it would just be the default (assigned when we create the `game` struct in `New()`). We could then override it in the tests by assigning our fake reader to the `game`.

Reply:
Removed the struct in the code during our session and replaced with separate parameters.